op-challenger/game/fault/solver/solver.go

```diff
@@ -97,22 +97,21 @@ func (s *claimSolver) AttemptStep(ctx context.Context, game types.Game, claim ty
 	if err != nil {
 		return StepData{}, err
 	}
-	index := claim.TraceIndex(s.gameDepth)
 	var preState []byte
 	var proofData []byte
 	var oracleData *types.PreimageOracleData
 
 	if !claimCorrect {
 		// Attack the claim by executing step index, so we need to get the pre-state of that index
-		preState, proofData, oracleData, err = s.trace.GetStepData(ctx, index)
+		preState, proofData, oracleData, err = s.trace.GetStepData(ctx, claim.Position)
 		if err != nil {
 			return StepData{}, err
 		}
 	} else {
 		// We agree with the claim so Defend and use this claim as the starting point to execute the step after
 		// Thus we need the pre-state of the next step
-		// Note: This makes our maximum depth 63 because we need to add 1 without overflowing.
-		preState, proofData, oracleData, err = s.trace.GetStepData(ctx, index+1)
+		defendingPosition := types.NewPosition(s.gameDepth, int(claim.TraceIndex(s.gameDepth))+1)
```

Comment:
I'd push this logic into `Position` then it can just increment the index at depth directly without having to tie it to `TraceIndex`:

```go
func (p Position) StepDefend() Position {
	return Position{
		depth:        p.depth,
		indexAtDepth: new(big.Int).Add(p.IndexAtDepth(), big.NewInt(1)),
	}
}
```

Maybe it should just be called `MoveRight` since step action really only makes sense at the maximum depth and this is agnostic of that.

Reply:
er, except in this PR indexAtDepth isn't a big.Int yet - I had your other PR still checked out. 😄

Reply:
But you can't ever perform a step against node 2 because it's not a leaf node. If it was, the node you'd want to wind up on would be 3. We don't need to consider moving down the tree here because to perform a step we must already be at the bottom of the tree.  Which is another good reason not to call it `StepDefend` really.

Reply:
Ah in this case, it doesn't make too much of a difference since the claim is checked to be a leaf node - I see. Will revert to your suggestion and use a `MoveRight` method using the claim depth 😄